refactor: LEAP-1907: Migrate EnterpriseBadge to UI library

Gondragos opened the pull request:
Moved `EnterpriseBadge` from the application codebase to the shared UI library for better reusability and modularity. Updated relevant imports and added a Storybook story for the badge. This change also includes moving the associated CSS and related IconSpark to the UI library and usage of scss modules.

<img width="1050" alt="image" src="https://github.com/user-attachments/assets/9152e120-9210-4066-a3f0-b2f10db061c0" />

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

Gondragos committed:
refactor: LEAP-1907: Migrate EnterpriseBadge to UI library

Moved EnterpriseBadge from the application codebase to the shared UI library for better reusability and modularity. Updated relevant imports and added a Storybook story for the badge. This change also includes moving the associated CSS and related IconSpark to the UI library and usage of scss modules.

Gondragos committed:
Merge branch 'develop' into fb-LEAP-1907

# Conflicts:
#	web/apps/labelstudio/src/pages/CreateProject/CreateProject.jsx

nass600 commented on web/libs/ui/src/lib/enterprise-badge/enterprise-badge.module.scss:
```diff
@@ -0,0 +1,35 @@
+  border-radius: 4px;
+  background: linear-gradient(135deg, #FFA663 0%, #FF7557 51.56%, #E37BD3 100%);
+  vertical-align: middle;
+  margin: 0 8px;
```
Margins should not be defined by atomic components, I will delegate this responsibility to higher components or the scaffolding given that depending on where to place it there might be different spacing needs

Gondragos replied:
That makes sense. But right now I'm just moving the existing component to the library. Somehow we already had these styles in it https://github.com/HumanSignal/label-studio/pull/7173/files#diff-a8632dfbab038917426ca3becd12f72a8d553418e432ca6667c66eb66aff7651L6

nass600 replied:
I believe, considering the few instances where we are [using it](https://github.com/search?q=org%3AHumanSignal+-is%3Aarchived+EnterpriseBadge&type=code), it might be worth to change it now so we ensure we have components that are not interacting outside their bounds in the UI and change the ones where it is being used.

nass600 reviewed (changes requested):
Apart from the margin related comments looks good to me

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13726313177)
> Successfully merged: Already up to date.

Gondragos committed:
fix: Exclude margin from EnterpriseBadge component styles

Added a `className` prop to the `EnterpriseBadge` component to allow external styling. Updated usage instances in CreateProject and GeneralSettings pages to apply a margin class for proper spacing. Also removed hardcoded margin from the component's default styles.

MihajloHoma commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13760774983)
> Successfully merged:  11 files changed, 21 insertions(+), 17 deletions(-)

MihajloHoma committed:
Merge branch 'develop' into 'fb-LEAP-1907'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13760774983

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13768425847)
> Successfully merged:  create mode 100644 web/apps/labelstudio/src/utils/query-client.ts

Gondragos committed:
Merge branch 'develop' into 'fb-LEAP-1907'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13768425847
